dev/tasks/vcpkg-tests/cpp-build-vcpkg.bat

```diff
@@ -70,6 +70,7 @@ cmake -G "Visual Studio 16 2019" -A x64 ^
       -DARROW_WITH_SNAPPY=ON ^
       -DARROW_WITH_ZLIB=ON ^
       -DARROW_WITH_ZSTD=ON ^
+      -DGTest_SOURCE=SYSTEM ^
```

Comment:
Hmm, how does this work on Windows? "system" would mean system-provided packages on Linux, but I don't think those exist at all on Windows.

Reply:
cc @kou for advice.

Reply:
We should not do this. We should fix https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L80-L86 instead.

(This is not a MINOR task.)

Reply:
Ok I'll close this PR and I think this might be an appropriate [JIRA](https://issues.apache.org/jira/browse/ARROW-14560) to use? I would gladly take your suggestion on how to fix this @kou , should this test only be used in case we also use conda? As it seems to work with vcpkg,

Reply:
It seems that ARROW-14560 is another problem and it has been fixed in upstream.

Could you open a new JIRA issue for it?

How about just removing https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L80-L86 ? The latest may solve the problem described in the comment.

If the problem isn't solved yet, we should report it to the conda's GTest package. We can use `if(ARROW_DEPENDENCY_SOURCE STREQUAL "CONDA" AND MSVC AND "${GTest_SOURCE}" STREQUAL "")` condition as a workaround for this case.